zerver/lib/actions.py

```diff
@@ -777,6 +778,20 @@ def active_humans_in_realm(realm: Realm) -> Sequence[UserProfile]:
     return UserProfile.objects.filter(realm=realm, is_active=True, is_bot=False)
 
 
+def get_user_default_language(request: HttpRequest, realm: Realm) -> str:
+    browser_locale = get_browser_locale(request)
+    if browser_locale is not None:
```

Comment:
i think we should also check whether the locale is among one of the locales we support. Else we default to realm.default_lamguage.

Reply:
yes! implemented it in the updated changes.